app/abci.go

```diff
@@ -236,23 +251,57 @@ func (app *HeimdallApp) NewProcessProposalHandler() sdk.ProcessProposalHandler {
 		var executionPayload engine.ExecutionPayload
 		err = json.Unmarshal(metadata.MarshaledExecutionPayload, &executionPayload)
 		if err != nil {
+			app.currBlockChan <- nextELBlockCtx{height: req.Height,
```

Comment:
Why we don't simply return `app.currBlockChan <- nextELBlockCtx{height: req.Height, context: ctx}` in these cases ?

Reply:
Saves a bor RPC call.

Reply:
What will be consequences if the ParentHash is invalid? Proposed by malicious or corrupted proposer.

Reply:
Actually we can use the last block value stored in the keeper.

Reply:
Ideally thats what we should use, instead of fetching latest block number from erigon. Fetching of the latest block number from erigon was just a temporary workaround.